### PR TITLE
ci: use conventional commits and release-please

### LINKED ISSUE #21: Streamline release process
# Summary

Due to this being a public action we should look into an easier release process that accounts for major/minor version changes on conventional commits.

- Enforce squash merging only to enforce a linear history
- Use conventional commits
- Add release-please to maintain a CHANGELOG.md file within the repository itself and handle bumping versions by commit

Let me know what you think

### PR DESCRIPTION
**Pull request by hanscg**
Resolves #21

### CODE REVIEW

**Comment by hanscg** [comment]
@jarviliam Is there a way to test release-please?

File: .github/workflows/release_please.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+name: Release please
+
+on:
+  push:
+    branches:
+      - release-please
```
Comment (FurqanHabibi) [inline .github/workflows/release_please.yml:6]: So we're not using the main branch as the release branch?

Comment (hanscg) [reply .github/workflows/release_please.yml]: Ahh sorry I forgot to revert my test commit. Reverted!

**Review by jarviliam — APPROVED** [review]
lgtm 👍 

For testing you could run the cli tool with the dry-run option but you might as well just wait until you have a fix/feature to add to this 👍 
https://github.com/googleapis/release-please/blob/main/docs/cli.md

**Review by FurqanHabibi — APPROVED** [review]

**Pull request merged**